python/torch_mlir/eager_mode/torch_mlir_types.py

```diff
@@ -0,0 +1,133 @@
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Also available under a BSD-style license. See LICENSE.
+#
+# pylint: disable=no-member, no-name-in-module, invalid-name, missing-function-docstring, fixme
+"""
+The following defines a set of classes for converting
+types used by Python and PyTorch into MLIR types from the
+`torch` dialect.
+
+The expected use of this module is to create an instance
+of one of the classes below, and then calling the
+`to_mlir` method to generate the MLIR representation
+of the type.
+
+Information about what types are supported by each class
+can be found in docstrings of each of the classes.
+"""
+
+import abc
+from typing import Any, Optional, Iterable
+
+import torch
+from torch_mlir import ir
+
+class TorchMlirType(abc.ABC):
+    """
+    A `TorchMlirType` is an object that produces MLIR
+    types in the `torch` dialect. The only requirement
+    for a class to be a subclass of `TorchMlirType`  is
+    to define a `to_mlir(self, ir.Context) -> ir.Type`.
+    Each class is allowed to have different types of
+    __init__ methods depending on the information they
+    require to produce the given MLIR representation.
+    """
+    @abc.abstractmethod
+    def to_mlir(self, context: ir.Context) -> ir.Type:
+        pass
+
+class TorchTensorTypeError(Exception):
```

Comment:
This is really just part of the API of ir_building.py, right? Can you move it there?